monitoring/metric/counter.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metric
+
+import (
+	"context"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+)
+
+type Counter interface {
+	Add(n int64)
+}
+type counter struct {
+	mu              sync.Mutex
+	name            string
+	value           int64
+	lastDumped      time.Time
+	lastDumpedValue int64
+}
+
+type safeMetrics struct {
+	mu sync.Mutex
+	m  map[string]*counter
+}
+
+var (
+	metrics = safeMetrics{
+		m: make(map[string]*counter),
+	}
+)
+
+func (m *counter) Add(n int64) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	m.value += n
+}
+
+func NewCounter(name string) Counter {
+	c := counter{name: name, lastDumped: time.Now()}
```

Comment:
The time aspect of this is somewhat related to having a very simplistic (as in, mostly inlined) "backend", dumping to the logs. Ideally, a more fleshed out backend would be keeping track of the time itself, possibly?

Keep an eye out for further developments... :smiley: